packages/react-components/src/queries/useLatestAssetPropertyValues/requestExecution/getLatestAssetPropertyValueRequest.ts

```diff
@@ -0,0 +1,36 @@
+import { GetAssetPropertyValue } from '@iot-app-kit/core';
+import { LatestAssetPropertyValueRequest } from '../types';
+
+export class GetLatestAssetPropertyValueRequest {
+  readonly #getAssetPropertyValue: GetAssetPropertyValue;
+
+  constructor({
+    getAssetPropertyValue,
+  }: {
+    getAssetPropertyValue: GetAssetPropertyValue;
+  }) {
+    this.#getAssetPropertyValue = getAssetPropertyValue;
+  }
+
+  public async send(
+    request: LatestAssetPropertyValueRequest,
+    settings: { abortSignal: AbortSignal }
+  ) {
+    try {
+      return await this.#getAssetPropertyValue(request, settings);
+    } catch (error) {
+      this.#handleError(request, error);
+    }
+  }
+
+  #handleError(
```

Comment:
Is this something we want to standardize across our hooks? Any reason for this more precise error handling?

Reply:
Yes, it's been the pattern in our request execution handling. This is also used to handle general errors across the entire batch or for a single error entry in the batch result.